salt/dex/dex.yaml

```diff
@@ -0,0 +1,136 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: dex
+  namespace: kube-system
+  labels:
+    kubernetes.io/cluster-service: "true"
+---
+# Map the LDAP Administrators role to the Kubernetes system:masters group
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: system:dex
+subjects:
+- kind: ServiceAccount
+  name: dex
+  namespace: kube-system
+roleRef:
+  kind: ClusterRole
+  name: cluster-admin
+  apiGroup: rbac.authorization.k8s.io
+---
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: dex
+  namespace: kube-system
+data:
+  config.yaml: |
+    issuer: "https://{{ pillar['api']['server']['external_fqdn'] }}:{{ pillar['dex']['node_port'] }}"
+    storage:
+      type: kubernetes
+      config:
+        inCluster: true
+    web:
+      https: 0.0.0.0:5556
+      tlsCert: /etc/dex/tls/dex.crt
+      tlsKey: /etc/dex/tls/dex.key
+    connectors:
+    - type: ldap
+      id: ldap
+      name: LDAP
+      config:
```

Comment:
I think we'll need to make most of this block configurable to allow for integrating with a pre-existing LDAP instance.. However, I think that should come as a follow up PR - this is large enough already.